pkg/argocd/repo.go

```diff
@@ -56,7 +52,30 @@ func CloneRepo(
 
 // -----------------------------------------------------------------------------
 
-func getRepoCreds(
+func GetKubeclientAndRepoCreds(
```

Comment:
Can this be split into two functions, each having a single responsibility?

Reply:
The whole point of the utility function was to eliminate repetition at each of the call sites, and at least one of them does require the kubeclient, so that's why the function combines the two.